modules/externalipam/netbox/netbox_client.rb

```diff
@@ -0,0 +1,201 @@
+require 'yaml'
+require 'json'
+require 'net/http'
+require 'uri'
+require 'sinatra'
+require 'externalipam/externalipam'
+require 'externalipam/ipam_helper'
+require 'externalipam/ipam_validator'
+require 'externalipam/api_resource'
+require 'externalipam/ip_cache'
+
+module Proxy::Netbox
+  # Implementation class for External IPAM provider Netbox
+  class NetboxClient
+    include Proxy::Log
+    include Proxy::Ipam::IpamHelper
+    include Proxy::Ipam::IpamValidator
+
+    def initialize(conf)
+      @api_base = "#{conf[:url]}/api/"
+      @token = conf[:token]
+      @api_resource = Proxy::Ipam::ApiResource.new(api_base: @api_base, token: "Token #{@token}")
+      @ip_cache = Proxy::Ipam::IpCache.instance
+      @ip_cache.provider_name('netbox')
+    end
+
+    def get_ipam_subnet(cidr, group_name = nil)
+      if group_name.nil? || group_name.empty?
+        get_ipam_subnet_by_cidr(cidr)
+      else
+        group_id = get_group_id(group_name)
+        get_ipam_subnet_by_group(cidr, group_id)
+      end
+    end
+
+    def get_ipam_subnet_by_group(cidr, group_id)
+      params = URI.encode_www_form({ status: 'active', prefix: cidr, vrf_id: group_id })
+      response = @api_resource.get("ipam/prefixes/?#{params}")
+      json_body = JSON.parse(response.body)
+      return nil if json_body['count'].zero?
+      subnet = subnet_from_result(json_body['results'][0])
+      return subnet if json_body['results']
```

Comment:
if `json_body['results']` was `nil`, the previously line would already have raised an exception
```suggestion
      subnet_from_result(json_body['results'][0])
```

This is a repeated pattern so I'm not commenting it on every instance.